### PR TITLE
[Feature] Tabs

### PR DESCRIPTION
**Pull request by nerdlet**
## Description

Fixes #178808450

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Screenshot 2021-08-06 at 14 54 46

Screenshot 2021-08-06 at 14 54 39

Screenshot 2021-08-06 at 14 54 34

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/components/StoriesNavigation/index.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      <div className={classes.tabsContent}>
+        <Tabs
+          value={value}
+          onChange={handleChange}
+          aria-label="simple tabs"
+          TabIndicatorProps={{
+            style: {
+              display: "none",
+            },
+          }}
```
Comment (KhadijaMahanga) [inline src/components/StoriesNavigation/index.js:108]: We can pass this through classes, `indicator` is the class to override I believe

File: src/components/StoriesNavigation/index.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+          />
+          <Tab
+            label={secondLabel}
+            {...a11yProps(1)}
+            disableRipple
+            classes={{
+              root: classes.secondTab,
+              wrapper: classes.wrapper,
+            }}
+          />
```
Comment (KhadijaMahanga) [inline src/components/StoriesNavigation/index.js:130]:
```suggestion
       {items.map((item, index) => (
          <Tab
            key={item.label}
            label={item.label}
            {...a11yTabProps(index, name)}
            classes={{ root... wrapper.... }} //we can use clsx here for active/ non-active
          />
        ))}
```

File: src/components/StoriesNavigation/index.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        </Tabs>
+        <Divider className={classes.divider} />
+      </div>
+      <div className={classes.tabpanel}>
+        <TabPanel value={value} index={0}>
+          {firstChild}
+        </TabPanel>
+        <TabPanel value={value} index={1}>
+          {secondChild}
+        </TabPanel>
```
Comment (KhadijaMahanga) [inline src/components/StoriesNavigation/index.js:140]: we can use map here too

File: src/components/StoriesNavigation/TabPanel.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    <Grid
+      container
+      direction="column"
+      role="tabpanel"
+      hidden={value !== index}
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
+      {value === index && <div>{children}</div>}
```
Comment (KhadijaMahanga) [inline src/components/StoriesNavigation/TabPanel.js:16]: We have already set `hidden = {value !== index}`, There's no need of checking again here

File: src/components/StoriesNavigation/TabPanel.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    </Grid>
+  );
+}
+
+TabPanel.propTypes = {
+  children: PropTypes.arrayOf(PropTypes.shape({})),
```
Comment (KhadijaMahanga) [inline src/components/StoriesNavigation/TabPanel.js:22]: `PropTypes.node`

**Review by KhadijaMahanga — COMMENTED** [review]
@nerdlet  👍🏾  Base for this PR should be main not `feature/news-insight`

**Review by kilemensi — COMMENTED** [review]
👍🏾 

-----

1. ~StoriesNavigation~Tabs: This same component is used in Data page as well.
2. Tab appear to have padding/margin to the left. They shouldn't (design vs implementation)
S

**Comment by nerdlet** [comment]
> 👍🏾
> 
> 1. ~StoriesNavigation~Tabs: This same component is used in Data page as well.
> 2. Tab appear to have padding/margin to the left. They shouldn't (design vs implementation)
>    S
@kilemensi Okay..I think I'll name it `NavigationTabs` instead. Naming it `Tabs`   alone seems to produce an eslint error due to material ui tabs component

**Comment by kilemensi** [comment]
>    @kilemensi Okay..I think I'll name it `NavigationTabs` instead. Naming it `Tabs`   alone seems to produce an eslint error due to material ui tabs component
>

Nah @nerdlet , when you're importing Material UI `Tabs`, give it a local name e.g. `import { Tabs as MuiTabs } from "@material-ui/core"`. We should then be able to use `Tabs`.

File: package.json [hunk 6]
```diff
@@ -27,6 +27,7 @@
     "@material-ui/core": "^4.12.1",
     "@material-ui/utils": "^4.11.2",
     "@svgr/webpack": "^5.5.0",
+    "classnames": "^2.3.1",
     "clsx": "^1.1.1",
```
Comment (kilemensi) [inline package.json:31]: We use `clsx` instead of `classnames`

File: src/components/Tabs/index.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    "&$selected": {
+      color: palette.primary.main,
+      fontWeight: 600,
+      fontSize: "16px",
+    },
+    "&:focus": {
+      color: palette.primary.main,
+    },
+  },
+  secondTab: {
```
Comment (kilemensi) [inline src/components/Tabs/index.js:51]: There is no first or 2nd tab, just tab

File: src/components/Tabs/TabPanel.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+function TabPanel({ children, value, index, ...other }) {
+  return (
+    <Grid
+      container
+      direction="column"
+      role="tabpanel"
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
```
Comment (kilemensi) [inline src/components/Tabs/TabPanel.js:14]: 1. Why `Grid`?
2. Aren't we setting `hidden` as well?

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: src/components/Tabs/TabPanel.js (hunk 8 above)
Comment (nerdlet) [reply src/components/Tabs/TabPanel.js]: @kilemensi Changed the above. For the hidden prop,  I did ran into an unusual behaviour where both all values in tabpanel were being shown whether or not the values were hidden. Removed it to test out why and forgot to add it again once i managed to fix it

**Review by kilemensi — APPROVED** [review]
🚀

File: src/components/Tabs/index.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+        onChange={handleChange}
+        aria-label="simple tabs"
+        classes={{
+          root: classes.tabs,
+          indicator: classes.indicator,
+        }}
+      >
+        {items.map((item, index) => (
+          <Tab
+            label={item.label}
```
Comment (KhadijaMahanga) [inline src/components/Tabs/index.js:41]:
```suggestion
        {items.map((item, index) => (
          <Tab
            key={item.label}
            label={item.label}
```

**Review by KhadijaMahanga — APPROVED** [review]

**Pull request merged**